crates/sui-faucet/src/faucet/write_ahead_log.rs

```diff
@@ -103,6 +105,18 @@ impl WriteAheadLog {
         }
         Ok(())
     }
+
+    pub(crate) fn set_in_flight(
+        &mut self,
+        coin: ObjectID,
+        bool: bool,
+    ) -> Result<(), TypedStoreError> {
+        if let Some(mut entry) = self.log.get(&coin)? {
+            entry.in_flight = bool;
+            self.log.insert(&coin, &entry)?;
+        }
+        Ok(())
```

Comment:
It's probably worth logging (or returning an error code that we can use to log further up in the stack) when we try and set someting in flight that wasn't in the WAL to begin with -- that seems like a bug.

Reply:
Given you're now logging this at the level of the faucet, it would be better to just encode a failure to find the entry as an error of some kind, to avoid double logging it, and to leverage the fact that at the faucet level you have more context (the recipient and UUID that I mentioned in an earlier comment).